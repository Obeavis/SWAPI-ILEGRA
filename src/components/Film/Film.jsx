import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { Col } from "react-bootstrap";
import Moment from 'react-moment';

const Film = (props) => (
	<Col lg={6} xl={4} className="mb-5" onClick={() => props.history.push(`/films/${props.film.episodeID}`)} >
		<div className="box-films box">
			<img
				alt={props.film.title}
				src={props.image}
				className="img-fluid"
			/>
			<span className="span-box">
				<span className="thumb-title text-bold text-center">{props.film.title}</span>
				<span className="thumb-date text-1 text-bold">
					<Moment format="LL">
						{new Date(props.film.releaseDate)}
					</Moment>
				</span>
			</span>
		</div>
	</Col>
);

Film.propTypes = {
	film: PropTypes.object
};

export default withRouter(Film);