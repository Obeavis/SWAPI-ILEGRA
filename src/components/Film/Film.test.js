import React from "react";
import { MemoryRouter } from 'react-router-dom';
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import Moment from "react-moment"
import Film from "./Film";

configure({ adapter: new Adapter() });

describe("Film Component", () => {

	const film = {
		episodeID: 4,
		releaseDate: "1977-05-25",
		title: "A New Hope"
	};

	const propsImage = require(`../../assets/images/films/episode-4.jpg`);

	const component = (

		<MemoryRouter initialEntries={['/films']}>
			<Film film={film} image={propsImage} />
		</MemoryRouter>
	)

	it("matches the snapshot", () => {
		const tree = renderer.create(component).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("Renders the right image", () => {
		const wrapper = mount(component);
		const image = wrapper.find('img').prop("src");

		expect(image).toEqual('episode-4.jpg');

	});

	it("Have the right title inside", () => {
		const wrapper = mount(component);
		const title = wrapper.find('.thumb-title').text();

		expect(title).toEqual('A New Hope');
	});

	it("Have the right date inside", () => {
		const wrapper = mount(component);
		const releaseDate = wrapper.find('.thumb-date').find('time').text();

		const date = mount(
			<Moment format="LL">
				{new Date(film.releaseDate)}
			</Moment>
		).find('time').text();

		expect(releaseDate).toEqual(date);
	});

	it("Route to the right path after click", () => {
		const wrapper = mount(component);
		const pathnameBeforeClick = wrapper.find('Router').props().history.location.pathname;
		expect(pathnameBeforeClick).toEqual('/films');

		wrapper.find('Col').simulate('click');

		const pathnameAfterClick = wrapper.find('Router').props().history.location.pathname;
		expect(pathnameAfterClick).toEqual('/films/4');
	});

})