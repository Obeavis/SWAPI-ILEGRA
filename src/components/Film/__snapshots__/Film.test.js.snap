// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Component matches the snapshot 1`] = `
<div
  className="mb-5 col-xl-4 col-lg-6"
  onClick={[Function]}
>
  <div
    className="box-films box"
  >
    <img
      alt="A New Hope"
      className="img-fluid"
      src="episode-4.jpg"
    />
    <span
      className="span-box"
    >
      <span
        className="thumb-title text-bold text-center"
      >
        A New Hope
      </span>
      <span
        className="thumb-date text-1 text-bold"
      >
        <time
          dateTime={"1977-05-25T00:00:00.000Z"}
        >
          May 24, 1977
        </time>
      </span>
    </span>
  </div>
</div>
`;
